src/components/exercise/ExerciseQuestionBlock.tsx

```diff
@@ -0,0 +1,235 @@
+import { Button, Icon } from '@chakra-ui/react'
+import React, { memo, useState } from 'react'
+import { defineMessages, useIntl } from 'react-intl'
+import styled, { css } from 'styled-components'
+import { ReactComponent as CheckCircleIcon } from '../../images/checked-circle.svg'
+import { ReactComponent as ErrorCircleIcon } from '../../images/error-circle.svg'
+import { ReactComponent as TickIcon } from '../../images/tick.svg'
+import { CommonLargeTextMixin, CommonTextMixin } from '../common'
+
+const messages = defineMessages({
+  prevQuestion: { id: 'program.ui.prevQuestion', defaultMessage: '上一題' },
+  nextQuestion: { id: 'program.ui.nextQuestion', defaultMessage: '下一題' },
+  submit: { id: 'program.ui.submit', defaultMessage: '送出' },
+  showResult: { id: 'program.ui.showResult', defaultMessage: '查看分數' },
+  correctAnswer: { id: 'program.status.correctAnswer', defaultMessage: '答案正確' },
+  errorAnswer: { id: 'program.status.errorAnswer', defaultMessage: '答案錯誤' },
+  correct: { id: 'program.status.correct', defaultMessage: '正解' },
+})
+
+const StyledQuestionCount = styled.span`
+  ${CommonTextMixin}
+`
+const StyledQuestion = styled.div`
+  font-size: 16px;
+  font-weight: 500;
+  letter-spacing: 0.2px;
+  color: var(--gray-darker);
+`
+const StyledDetail = styled.div`
+  padding: 24px;
+  border-radius: 4px;
+  background-color: var(--gray-lighter);
+`
+const StyledDetailTitle = styled.h4`
+  display: inline-block;
+  font-size: 16px;
+  font-weight: bold;
+  letter-spacing: 0.2px;
+  color: var(--gray-darker);
+`
+const StyledDetailContent = styled.p`
+  ${CommonLargeTextMixin}
+`
+
+const ExerciseQuestionBlock: React.FC<{
+  showDetail: boolean
+  exercises: {
+    options: {
+      answer: string
+      isAnswer: boolean
+      isSelected: boolean
+    }[]
+    question: string
+    detail: string
+    score: number
+  }[]
+  onSetStatusResult: () => void
```

Comment:
NITS: it is better to use semantic word like `onSeeResult()`, `onFinish()` or `onCheckScore()`